### PR TITLE
feat(FEC-7140): seekbar thumbs sprite from config

### PR DESCRIPTION
**Pull request by dvh91**
### Description of the Changes

Seekbar thumbs should be agnostic and get stripe image url, slices count and slice width from config.
config sample of kaltura-player-js:
```
let config = {
 ...
 ui: {
      SeekBarPlaybackContainer: {
        thumbsSprite: 'http://cdntesting.qa.mkaltura.com/p/1091/sp/109100/thumbnail/entry_id/0_wifqaipd/version/100042/height/90/width/160/vid_slices/200',
        thumbsWidth: 160,
        thumbsSlices: 200
      }
    }
}
```

### CheckLists

- [x] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [x] test are passing in local environment 
- [x] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/components/base.js [hunk 1]
```diff
@@ -30,6 +30,20 @@ class BaseComponent extends Component {
...
   }
+
+  /**
+   * Gets config param value
+   * @param {string} property property name
+   * @returns {string} config value
+   */
+  getConfig(property: string): ?string {
+    try {
+      return this.config.components[this.name][property];
```
Comment (OrenMe) [inline src/components/base.js]: isn't it better to solve the config object on init then to try-catch on each access?
IMO - better init the config in the constructor once with validation that it exist and then have the getConfig just return the property
```javascript
   constructor(obj?: Object = { config: {} }) {
     ...
     this.config = (obj.config.components && obj.config.components[this.name]) || {};
      ...
    }
```
```javascript
getConfig(property: string): ?string {
   return this.config[property];
}
```

this way a component will also carry only its config and not all components config

**Comment by OrenMe** [comment]
@dvh91 were you able to look at my previous comment? how can we enable config changes on the fly?

**Comment by dvh91** [comment]
@OrenMe Yes. I finalise the solution now. basically the solution is to expose new uimanager 'updateConfig' method save config obj in the store. then consume this config when needed in components. once the config will be changed in store, the component that use the relevant config will be rerendered.

**Comment by dvh91** [comment]
`kaltura-player.js` required changes:
```
  constructor(player: Player, targetId: string, config: Object, uiConfig: ?Object) {
    ...
    return {
      ...
      setConfig: this.setConfig.bind(this)
    }
  }

  setConfig(config: Object, componentAlias?: string) {
    this._uiManager.setConfig(config, componentAlias);
  }
```

File: src/components/seekbar/seekbar.js [hunk 2]
```diff
@@ -275,9 +277,12 @@ class SeekBarControl extends Component {
   getTimeBubbleOffset(): number {
     if (this._timeBubbleElement) {
       let leftOffset = (this.state.virtualTime / this.props.duration * this._seekBarElement.clientWidth) - (this._timeBubbleElement.clientWidth / 2);
-      if (leftOffset < 0) return 0;
-      else if (leftOffset > this._seekBarElement.clientWidth - this._timeBubbleElement.clientWidth) return (this._seekBarElement.clientWidth - this._timeBubbleElement.clientWidth);
-      else return leftOffset;
+      if (leftOffset < 0)
```
Comment (OrenMe) [inline src/components/seekbar/seekbar.js]: add  `{}` in all conditions

**Pull request merged**